src/main/java/org/betonquest/betonquest/modules/logger/custom/ChatLogFormatter.java

```diff
@@ -25,11 +25,12 @@ public ChatLogFormatter() {
 
     @Override
     public String format(final LogRecord record) {
-        final String color = formatColor(record);
-        final boolean isBQ = record instanceof BetonQuestLogRecord;
-        final BetonQuestLogRecord logRecord = isBQ ? (BetonQuestLogRecord) record : null;
-        final String plugin = isBQ && !logRecord.getPlugin().isEmpty() ? logRecord.getPlugin() : "";
-        final String questPackage = isBQ && !logRecord.getPack().isEmpty() ? "<" + logRecord.getPack() + "> " : "";
+        if (!(record instanceof BetonQuestLogRecord logRecord)) {
+            throw new UnsupportedOperationException("The ChatLogFormatter can only format BetonQuestLogRecords!");
+        }
```

Comment:
This is a Liskov substitution principle violation. It has to be either fixed or at least documented.